### PR TITLE
✏️  511 adding the types coming from the openapi specs

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the new types to match the backend open API specs.

## Changes

- Introduces an auto generator for typescript.
- Updates models and factories to use the new format.
- Updates components, services, and redux to use the new data format.

## Fixes

- https://github.com/kubeshop/tracetest/issues/511

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
     "test:unit": "craco test --watchAll=false --coverage",
     "test:integration": "npm run cy:run",
     "lint": "npm run eslint && npm run types",
+    "types:generate": "openapi-typescript ../api/openapi.yaml --output ./src/types/Generated.types.ts",
```
Comment (xoscar) [inline web/package.json:49]: New autogenerator command

File: web/src/components/AssertionCheckRow/AssertionCheckRow.tsx [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
   onSelectSpan,
 }) => {
-  const signatureSelectorList = signature.map(({value}) => value).concat([`#${spanId.slice(-4)}`]) || [];
+  const {
+    run: {trace},
+  } = useTestRun();
+  const span = useMemo(() => trace?.spans.find(({id}) => id === spanId), [spanId, trace?.spans]);
+
+  const signatureSelectorList = span?.signature.map(({value}) => value).concat([`#${spanId.slice(-4)}`]) || [];
   const spanLabelList = difference(signatureSelectorList, assertionSelectorList);
```
Comment (xoscar) [inline web/src/components/AssertionCheckRow/AssertionCheckRow.tsx:29]: Using new format

File: web/src/components/AssertionForm/AssertionFormProvider.tsx [hunk 3]
```diff
@@ -48,34 +44,10 @@ const AssertionFormProvider: React.FC<{testId: string}> = ({children, testId}) =
...
-          },
-        });
-      }
-
-      setIsOpen(false);
-    },
-    [createAssertion, formProps, testId, updateAssertion]
-  );
+  const onSubmit = useCallback(({selectorList, assertionList}: IValues) => {
+    console.log('@@onSubmit', selectorList, assertionList);
```
Comment (xoscar) [inline web/src/components/AssertionForm/AssertionFormProvider.tsx:48]: This is going to use the new backend logic

File: web/src/models/AssertionResult.model.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import {TAssertionResult, TRawAssertionResult} from '../types/Assertion.types';
```
Comment (xoscar) [inline web/src/models/AssertionResult.model.ts:1]: New models

File: web/src/types/Assertion.types.ts [hunk 5]
```diff
@@ -1,72 +1,43 @@
 import {CompareOperator} from 'constants/Operator.constants';
-import {LOCATION_NAME} from 'constants/Span.constants';
```
Comment (xoscar) [inline web/src/types/Assertion.types.ts:2]: I wanted to use interfaces but I found out those don't support property lookup so I decided to use types instead.

**Review by kdhamric — APPROVED** [review]
I skimmed through it - big change and a lot of work! (but understand why)

**Review by cescoferraro — APPROVED** [review]
LGTM

**Pull request merged**